server/routes/ad.js

```diff
@@ -19,60 +19,73 @@ router.put('/', (req, res, next) => {
     email,
     zipCode,
   } = req.body;
-  sheets.spreadsheets.values.get(
-    {
-      auth: jwtClient,
-      spreadsheetId: spreadsheetId,
-      range: sheetRange,
-    },
-    (err, response) => {
-      if (err) {
-        console.log(err);
-      } else {
-        console.log('Ad retrieved from ' + sheetRange);
-        const ads = response.data.values;
-        let rowCounter = 2;
-        for (let ad of ads) {
-          if (ad[0] === req.body.id) {
-            let values = [
-              [
-                req.body.id,
-                title,
-                price,
-                description,
-                photo,
-                condition,
-                email,
-                zipCode,
-                modifiedDate,
-              ],
-            ];
-            const resource = { values };
-            return sheets.spreadsheets.values.update(
-              {
-                auth: jwtClient,
-                spreadsheetId: spreadsheetId,
-                range: `Ads!A${rowCounter}:J${rowCounter}`,
-                valueInputOption: 'USER_ENTERED',
-                resource,
-              },
-              (err, response) => {
-                if (err) {
-                  console.log(err);
-                } else {
-                  console.log('ad');
-                  const ads = response.data.values;
-                  return res.status(204).json();
-                }
-              }
-            );
-          }
-          rowCounter++;
-        }
-        return res.status(404).json('Ad Not Found. Please check ID.');
+
+  const sheetsReq = {
+    auth: jwtClient,
+    spreadsheetId: spreadsheetId,
+  };
+
+  // Helper method for processing data
+  const findRowById = async (rows, id) => {
+    let processedData = null;
+
+    rows.forEach((row, index) => {
+      // Match found
+      if (row[0] === id) {
+        // Row is found at index + 2 to account for first row of column headers
+        const row = index + 2;
+
+        const values = [
+          [
+            id,
+            title,
+            price,
+            description,
+            photo,
+            condition,
+            email,
+            zipCode,
+            modifiedDate,
+          ],
+        ];
+
+        processedData = {
+          resource: { values },
+          sheetRange: `Ads!A${row}:J${row}`,
+        };
       }
+    });
+
+    return processedData;
+  };
+
+  try {
+    // GET all rows in sheet
+    const getRes = await sheets.spreadsheets.values.get({
```

Comment:
@raymond-sde , so `sheets.spread.values.get()` returns a Promise? Is this a situation where you need to resolve it, either by 1) passing a callback (which I did), 2) using a .then() block, or 3) `await` the Promise?